ecosystem/typescript/sdk_v2/src/transactions/types/typeTag.ts

```diff
@@ -0,0 +1,252 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable max-classes-per-file */
+import { AccountAddress } from "../../core";
+import { Deserializer, Serializable, Serializer } from "../../bcs";
+import { Identifier } from "./identifier";
+import { TypeTagParser } from "./typeTagParser";
+import { TypeTagVariants } from "../../types";
+
+export abstract class TypeTag extends Serializable {
+  abstract serialize(serializer: Serializer): void;
+
+  static deserialize(deserializer: Deserializer): TypeTag {
+    const index = deserializer.deserializeUleb128AsU32();
+    switch (index) {
+      case TypeTagVariants.Bool:
+        return TypeTagBool.load(deserializer);
+      case TypeTagVariants.U8:
+        return TypeTagU8.load(deserializer);
+      case TypeTagVariants.U64:
+        return TypeTagU64.load(deserializer);
+      case TypeTagVariants.U128:
+        return TypeTagU128.load(deserializer);
+      case TypeTagVariants.Address:
+        return TypeTagAddress.load(deserializer);
+      case TypeTagVariants.Signer:
+        return TypeTagSigner.load(deserializer);
+      case TypeTagVariants.Vector:
+        return TypeTagVector.load(deserializer);
+      case TypeTagVariants.Struct:
+        return TypeTagStruct.load(deserializer);
+      case TypeTagVariants.U16:
+        return TypeTagU16.load(deserializer);
+      case TypeTagVariants.U32:
+        return TypeTagU32.load(deserializer);
+      case TypeTagVariants.U256:
+        return TypeTagU256.load(deserializer);
+      default:
+        throw new Error(`Unknown variant index for TypeTag: ${index}`);
+    }
+  }
+}
+
+export class TypeTagBool extends TypeTag {
+  serialize(serializer: Serializer): void {
+    serializer.serializeU32AsUleb128(TypeTagVariants.Bool);
+  }
+
+  static load(_deserializer: Deserializer): TypeTagBool {
+    return new TypeTagBool();
+  }
```

Comment:
Am I missing something? Looks like deserializer is not being use here? (Same as other classes)

Reply:
I think this was just to implement a common interface where `load` takes a `deserializer`, but in this case, it isn't used, because there's no extra data in a serialized `TypeTagBool` other than the initial enum variant-

that is, there's only 1 byte when you serialize a `TypeTag`, and that's the enum variant. deserializing it results in a call from `deserialize` in `TypeTag` and then `load` in the corresponding type. Since there's nothing else to deserialize, `load` ignores the `deserializer` arg!